gnovm/pkg/gnolang/preprocess.go

```diff
@@ -237,33 +234,36 @@ func initStaticBlocks(store Store, ctx BlockNode, bn BlockNode) {
 
 		// ----------------------------------------
 		case TRANS_BLOCK:
+			pushInitBlock(n.(BlockNode), &last, &stack)
 			switch n := n.(type) {
-			case *BlockStmt:
-				pushInitBlock(n, &last, &stack)
-			case *ForStmt:
-				pushInitBlock(n, &last, &stack)
-			case *IfStmt:
-				pushInitBlock(n, &last, &stack)
 			case *IfCaseStmt:
-				pushInitRealBlock(n, &last, &stack)
 				// parent if statement.
 				ifs := ns[len(ns)-1].(*IfStmt)
 				// anything declared in ifs are copied.
 				for _, n := range ifs.GetBlockNames() {
 					last.Predefine(false, n)
 				}
 			case *RangeStmt:
-				pushInitBlock(n, &last, &stack)
 				if n.Op == DEFINE {
 					if n.Key != nil {
-						last.Predefine(false, n.Key.(*NameExpr).Name)
+						nx := n.Key.(*NameExpr)
+						if nx.Name != blankIdentifier {
+							// XXX, temp method to NOT heapDefine loopvar for range,
+							// to make it consistent with `for i:=0;i<3;i++;`,
+							// this should be uncommented when supporting full Go1.22 loopvar.
+							//nx.Type = NameExprTypeDefine
+							last.Predefine(false, nx.Name)
+						}
 					}
```

Comment:
temp method to `NOT` support loopvar of range, to make it consistent with for i:=0;i<3;i++.